compiler/plc_ast/src/pre_processor.rs

```diff
@@ -59,7 +61,7 @@ pub fn pre_process(unit: &mut CompilationUnit, mut id_provider: IdProvider) {
                 {
                     let name: &str = name.as_ref().map(|it| it.as_str()).unwrap_or("undefined");
 
-                    let type_name = typesystem::create_internal_type_name("", name);
+                    let type_name = format!("__{name}"); // TODO: Error prone if typesystem:create_internal_type_name ever gets changed
```

Comment:
I'm guessing this hardcoded for the same reason "VOID" is in `ast.rs`?

Reply:
Yup, and I'm not a big fan of it :P